### PR TITLE
Added test for updating repo's checksum type on capsule (BZ1288656)

### PR DESCRIPTION
**Pull request by abalakh**
https://bugzilla.redhat.com/show_bug.cgi?id=1288656
```python
py.test -v tests/foreman/api/test_contentmanagement.py -k test_positive_checksum_sync
============================= test session starts ==============================
platform darwin -- Python 2.7.13, pytest-3.1.3, py-1.4.34, pluggy-0.4.0 -- /Users/andrii/workspace/env/bin/python
cachedir: .cache
rootdir: /Users/andrii/workspace/robottelo, inifile:
plugins: xdist-1.20.1, services-1.2.1, mock-1.6.2, forked-0.2, cov-2.5.1
collected 10 items
2017-10-27 17:36:30 - conftest - DEBUG - Deselect of WONTFIX BZs is disabled in settings


tests/foreman/api/test_contentmanagement.py::CapsuleContentManagementTestCase::test_positive_checksum_sync PASSED

============================== 9 tests deselected ==============================
=================== 1 passed, 9 deselected in 163.15 seconds ===================
```

### CODE REVIEW

File: tests/foreman/api/test_contentmanagement.py [hunk 1]
```diff
@@ -217,6 +217,122 @@ def test_positive_uploaded_content_library_sync(self):
...
+        lce_repo_path = form_repo_path(
+            org=org.label,
+            lce=lce.label,
+            cv=cv.label,
+            prod=product.label,
+            repo=repo.label,
+        )
+        result = ssh.command(
+            'grep -o \'checksum type="sha1"\' {}'
+            .format(os.path.join(lce_repo_path, repomd_path)),
```
Comment (ldjebran) [inline tests/foreman/api/test_contentmanagement.py]: think we cannot use os.path.join here as this URL is a remote one and not a local one.
This code for example can be run from windows and the path separator will be different from the one on the target vm.

File: tests/foreman/api/test_contentmanagement.py [hunk 2]
```diff
@@ -217,6 +217,122 @@ def test_positive_uploaded_content_library_sync(self):
...
+            hostname=self.capsule_hostname
+        )
+        self.assertNotEqual(result.return_code, 0)
+        self.assertEqual(len(result.stdout), 0)
+        result = ssh.command(
+            'grep -o \'checksum type="sha256"\' {}'
+            .format(os.path.join(lce_repo_path, repomd_path)),
```
Comment (ldjebran) [inline tests/foreman/api/test_contentmanagement.py]: same comment as above.

File: tests/foreman/api/test_contentmanagement.py (hunk 1 above)
Comment (abalakh) [reply tests/foreman/api/test_contentmanagement.py]: Good point, thanks

**Comment by abalakh** [comment]
```python
py.test -v tests/foreman/api/test_contentmanagement.py -k test_positive_checksum_sync
============================= test session starts ==============================
platform darwin -- Python 2.7.13, pytest-3.1.3, py-1.4.34, pluggy-0.4.0 -- /Users/andrii/workspace/env/bin/python
cachedir: .cache
rootdir: /Users/andrii/workspace/robottelo, inifile:
plugins: xdist-1.20.1, services-1.2.1, mock-1.6.2, forked-0.2, cov-2.5.1
collected 10 items
2017-11-08 17:14:37 - conftest - DEBUG - Deselect of WONTFIX BZs is disabled in settings


tests/foreman/api/test_contentmanagement.py::CapsuleContentManagementTestCase::test_positive_checksum_sync PASSED

============================== 9 tests deselected ==============================
================== 1 passed, 9 deselected in 2271.64 seconds ===================
```

**Pull request merged**